pkg/metrics/pushclient/pushclient.go

```diff
@@ -0,0 +1,78 @@
+package pushclient
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/prometheus/common/expfmt"
+)
+
+// aggregationClient is a custom HTTP client for use with
+// WeaveWorks Prometheus Aggregation PushGateway
+// We will embed http.Client and essentially
+// "override" the Do function
+type aggregationClient struct {
+	// embed a stdlib http.Client
+	*http.Client
+}
+
+// gatewayURL is the default url that will be used as the endpoint to push to.
+// To push to a centralized prometheus aggregation gateway, change this url.
+var gatewayURL = "http://localhost:9092"
```

Comment:
this in between the aggregationClient struct and functions, move it out of the middle.